src/client.js

```diff
@@ -742,6 +742,20 @@ MatrixClient.prototype.getRoom = function(roomId) {
     return this.store.getRoom(roomId);
 };
 
+/**
+ * Preloads the member list for the given room id,
+ * in case lazy loading of memberships is in use.
+ * @param {string} roomId The room ID
+ */
+MatrixClient.prototype.loadRoomMembersIfNeeded = function(roomId) {
+    const room = this.getRoom(roomId);
+    if (!room || !room.membersNeedLoading()) {
+        return;
+    }
+    const membersPromise = this.joinedMembers(roomId);
+    room.setLazilyLoadedMembers(membersPromise);
```

Comment:
Because putting the flag (`_membersNeedLoading`) whether loading has started in Room felt natural as it needs to be kept per room. I agree it's a bit weird, but modifying the flag, private field on Room, from MatrixClient would also not be pretty IMO. I didn't put the whole thing in Room as the pattern seems to be to do network I/O calls in client only. Wdyt?

I could still return a promise from here though, and put the error logging in RoomView then. I'll do that.

Reply:
@dbkr To clarify how the flag is used to avoid a race, I commented here in the follow-up PR. Makes more sense now?:
https://github.com/matrix-org/matrix-js-sdk/pull/669/commits/6570f56538be8198e69ff4308b69f11c47511fb6